cnd/src/network/comit_ln/swaps.rs

```diff
@@ -0,0 +1,602 @@
+            .retain(|digest, _| !digests.contains(digest));
+    }
+
+    /// This does not test external behaviour but the aim is to ensure we are
+    /// not consuming memory for no reason.
+    #[cfg(test)]
+    fn swap_in_pending_hashmaps(&self, digest: &SwapDigest) -> bool {
+        self.pending_confirmation.get(digest).is_some()
+            || self.pending_announcement.get(digest).is_some()
+            || self.pending_creation.get(digest).is_some()
+            || self.timestamps.get(digest).is_some()
+    }
+}
+
+impl Default for Swaps<ReplySubstream<NegotiatedSubstream>> {
+    fn default() -> Self {
+        Swaps {
+            pending_confirmation: Default::default(),
+            pending_announcement: Default::default(),
+            pending_creation: Default::default(),
+            swaps: Default::default(),
+            swap_ids: Default::default(),
+            timestamps: Default::default(),
+        }
+    }
+}
+
+#[cfg(test)]
+impl Default for Swaps<()> {
+    fn default() -> Self {
+        Swaps {
+            pending_confirmation: Default::default(),
+            pending_announcement: Default::default(),
+            pending_creation: Default::default(),
+            swaps: Default::default(),
+            swap_ids: Default::default(),
+            timestamps: Default::default(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        asset,
+        asset::ethereum::FromWei,
+        identity,
+        network::DialInformation,
+        swap_protocols::{EthereumIdentity, Role},
+    };
+    use digest::Digest;
+
+    fn create_params() -> HanEtherereumHalightBitcoinCreateSwapParams {
+        HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Alice,
+            peer: DialInformation {
+                peer_id: PeerId::random(),
+                address_hint: None,
+            },
+            ethereum_identity: EthereumIdentity::from(identity::Ethereum::random()),
+            ethereum_absolute_expiry: 12345.into(),
+            ethereum_amount: asset::Ether::from_wei(9_001_000_000_000_000_000_000u128),
+            lightning_identity: identity::Lightning::random(),
+            lightning_cltv_expiry: 12345.into(),
+            lightning_amount: asset::Bitcoin::from_sat(1_000_000_000),
+        }
+    }
+
+    #[test]
+    fn created_swap_as_pending_confirmation_can_be_retrieved() {
+        let create_params = create_params();
+        let digest = create_params.clone().digest();
+        let local_swap_id = LocalSwapId::default();
+        let mut swaps = Swaps::<()>::default();
+
+        let creation = swaps.create_as_pending_confirmation(
+            digest,
+            local_swap_id.clone(),
+            create_params.clone(),
+        );
+
+        assert!(creation.is_ok());
+
+        let created_swap = swaps.get_created_swap(&local_swap_id);
+
+        assert!(created_swap.is_some());
+
+        assert_eq!(created_swap.unwrap(), create_params)
```

Comment:
Nit: Avoid newlines to embrace arrange-act-assert :)

```suggestion
        assert!(creation.is_ok());
        let created_swap = swaps.get_created_swap(&local_swap_id);
        assert!(created_swap.is_some());
        assert_eq!(created_swap.unwrap(), create_params)
```